pkg/broker/aws_sdk.go

```diff
@@ -80,3 +88,20 @@ func assumeTargetRole(sess *session.Session, params map[string]string, region st
 
 	return sessionTargetAccount, nil
 }
+
+func getObjectBody(s3svc S3Client, bucket string, key string) (body []byte, err error) {
+	obj, err := s3svc.Client.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		return nil, err
+	} else if obj.Body == nil {
```

Comment:
When testing initially I did manage to get a nil Body, but not sure what's needed to reproduce. I'm inclined to leave the check in in-place unless we can see a case where it could do harm.

Reply:
oops, I missed that we could do that already with the `-spec.yaml` implementation. Will add the functionality